src/scf/self_consistent_field.jl

```diff
@@ -125,6 +127,8 @@ Solve the Kohn-Sham equations with a SCF algorithm, starting at ρ.
     n_iter = 0
     energies = nothing
     ham = nothing
+    nonconsistent_energies = nothing
+    previous_energies = nothing
```

Comment:
I think this should not live here, much rather the convergence / printing functors should cache that by themselves.

Reply:
agree